src/overlays/actors/ovl_En_Trt2/z_en_trt2.c

```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
+        ret = true;
+    } else {
+        ret = false;
+    }
+    return ret;
+}
+
+s32 func_80AD4B4C(EnTrt2* this, GlobalContext* globalCtx) {
+    s32 sp24 = false;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        sp24 = true;
+        this->actor.speedXZ = 0.0f;
+        func_80AD349C(this);
+        this->unk_3B4 = this->unk_3B2;
+        if (this->actor.bgCheckFlags & 1) {
+            if ((player->transformation != PLAYER_FORM_HUMAN) && (player->transformation != PLAYER_FORM_FIERCE_DEITY)) {
+                this->unk_3A8 = 0x84F;
+            }
+            func_80AD3DA4(this, globalCtx);
+        } else if ((player->transformation == PLAYER_FORM_HUMAN) ||
+                   (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+            this->unk_3B2 = 16;
+            this->actor.velocity.y = 1.5f;
+        } else {
+            this->unk_3A8 = 0x84F;
+            this->unk_3D8 = 1;
+            func_80AD3380(&this->skelAnime, sAnimations, 7);
+            this->unk_3B2 = 7;
+        }
+    }
+    return sp24;
+}
+
+s32 func_80AD4C4C(EnTrt2* this) {
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.xzDistToPlayer < 100.0f) {
+            return true;
+        }
+    } else if ((this->actor.xzDistToPlayer > 50.0f) && (this->actor.xzDistToPlayer < 200.0f)) {
+        return true;
+    }
+    return false;
+}
+
+s32 func_80AD4CCC(EnTrt2* this, GlobalContext* globalCtx) {
+    s16 sp1E = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (((this->unk_3B2 == 4) || (this->unk_3B2 == 5)) && this->actor.isTargeted && !(this->actor.bgCheckFlags & 1) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
+        this->actor.speedXZ = 0.0f;
+        this->actor.velocity.y = 1.5f;
+        this->unk_3B2 = 16;
+        return true;
+    }
+
+    if (func_80AD4C4C(this) && this->actor.isTargeted && (sp1E < 0x4000) && (sp1E >= -0x3FFF)) {
+        func_800B863C(&this->actor, globalCtx);
+    }
+
+    return true;
+}
+
+void func_80AD4DB4(EnTrt2* this, GlobalContext* globalCtx) {
+    static Vec3f D_80AD5904 = { 0.0f, 50.0f, 0.0 };
+
+    this->actor.flags &= ~0x10;
+    Actor_SetObjectSegment(globalCtx, &this->actor);
+    Actor_SetScale(&this->actor, 0.008f);
+    this->path = func_8013D648(globalCtx, ENTRT2_GET_FC00(&this->actor), 0x3F);
+    this->unk_3AE = Rand_S16Offset(100, 50);
+    this->unk_3B0 = 10;
+    this->unk_3A8 = 0;
+
+    this->actor.world.pos.x += D_80AD5904.x;
+    this->actor.world.pos.y += D_80AD5904.y;
+    this->actor.world.pos.z += D_80AD5904.z;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 20.0f);
+    func_80AD469C(this, globalCtx);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.colChkInfo.cylRadius = 50;
+
+    this->unk_3C0 = 0;
+    this->unk_3D8 = 0;
```

Comment:
```suggestion
    this->unk_3D8 = false;
```
This variable is boolean